stan/math/prim/mat/prob/multi_normal_rng.hpp

```diff
@@ -33,14 +46,21 @@ namespace stan {
       check_symmetric(function, "Covariance matrix", S);
       check_finite(function, "Location parameter", mu);
 
+      Eigen::LLT<Eigen::MatrixXd> llt_of_S = S.llt();
+      if (llt_of_S.info() != Eigen::Success) {
+        throw std::domain_error("multi_normal_rng:"
```

Comment:
We've specialized check_pos_definite to accept an LLT object, so it won't
do the decomposition twice.


Rob

On Fri, Sep 2, 2016 at 1:12 PM, Bob Carpenter <notifications@github.com>
wrote:

> In stan/math/prim/mat/prob/multi_normal_rng.hpp
> <https://github.com/stan-dev/math/pull/379#discussion_r77380536>:
>
> > @@ -33,14 +46,21 @@ namespace stan {
> >        check_symmetric(function, "Covariance matrix", S);
> >        check_finite(function, "Location parameter", mu);
> >
> > +      Eigen::LLT<Eigen::MatrixXd> llt_of_S = S.llt();
> > +      if (llt_of_S.info() != Eigen::Success) {
> > +        throw std::domain_error("multi_normal_rng:"
>
> No! That causes the matrix to be factored twice. - Bob
> … <#m_-4239402204551773686_>
> On Sep 2, 2016, at 6:31 PM, Daniel Lee ***@***.***> wrote: In
> stan/math/prim/mat/prob/multi_normal_rng.hpp: > @@ -33,14 +46,21 @@
> namespace stan { > check_symmetric(function, "Covariance matrix", S); >
> check_finite(function, "Location parameter", mu); > > +
> Eigen::LLT<Eigen::MatrixXd> llt_of_S = S.llt(); > + if (llt_of_S.info() !=
> Eigen::Success) { > + throw std::domain_error("multi_normal_rng:" Want to
> just use check_pos_definite()? If so, you can replace lines 50-54 with:
> check_pos_definite("multi_normal_rng", "matrix argument", llt_of_S); If
> you're going down path, could you change the error message on
> check_pos_definite? What you've written is so much clearer than "Cholesky
> decomposition of <> failed". — You are receiving this because you authored
> the thread. Reply to this email directly, view it on GitHub, or mute the
> thread.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/stan-dev/math/pull/379/files/2b26c0bc41646f8b7991915463d8d22452485008#r77380536>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADvmhy2bQNXqls-_G_I4HuTtpFGq7N2_ks5qmFj3gaJpZM4Jz32z>
> .
>